app/full_text_search.py

```diff
@@ -113,6 +135,26 @@ def search(repo, query_txt, max_results):
     return results
 
 
+def create_fields_for_rank(field_name, value):
+    """
+    Creates fields for ranking.
+    Args:
+        field_name: field name
+        value: field value
+    Returns:
+        array of appengine_search.TextField(name=field_name, value=value)
+           (length: REPEAT_COUNT_FOR_RANK)
+    """
+    if not value:
+        return []
+
+    fields = []
+    for x in range(REPEAT_COUNT_FOR_RANK):
```

Comment:
Use xrange instead of range. They have the same effect in this context, but range(5) actually creates an array [0, 1, 2, 3, 4, 5] while xrange doesn't.